### ISSUE TITLE
Improve the Q&A Pairs Transformation Pipeline

### ISSUE THREAD

**Issue by grayJiaaoLi** [amosproj/amos2024ss08-cloud-native-llm#42]
## User story
1. As a developer
2. I want / need to refine the Q&A mechanism and fix its problems
3. So that the quality of training data can be ensured

## Acceptance criteria
* Fix the bugs and other observed problems
* Improve the Q&A pairs transformation pipeline
* Improve the quality of training data

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release
